appserver/payara-appserver-modules/payara-micro-service/src/main/java/fish/payara/appserver/micro/services/PayaraInstanceImpl.java

```diff
@@ -243,25 +240,29 @@ public void event(Event event) {
             PayaraInternalEvent pie = new PayaraInternalEvent(PayaraInternalEvent.MESSAGE.ADDED, me);
             ClusterMessage<PayaraInternalEvent> message = new ClusterMessage<>(pie);
             this.cluster.getEventBus().publish(INTERNAL_EVENTS_NAME, message);
-            
+
             for (String appName : appRegistry.getAllApplicationNames()) {
                 me.addApplication(new ApplicationDescriptorImpl(appRegistry.get(appName)));
             }
-            
+
             cluster.getClusteredStore().set(INSTANCE_STORE_NAME, myCurrentID, me);
             executor.scheduleAtFixedRate(() -> {
                 me.setLastHeartBeat(System.currentTimeMillis());
                 if (myCurrentID != null) {
                     cluster.getClusteredStore().set(INSTANCE_STORE_NAME, myCurrentID, me);
                 }
             }, 0, 5, TimeUnit.SECONDS);
-        } 
-        // Adds the application to the clustered register of deployed applications
+        } // Adds the application to the clustered register of deployed applications
         else if (event.is(Deployment.APPLICATION_STARTED)) {
             if (event.hook() != null && event.hook() instanceof ApplicationInfo) {
                 ApplicationInfo applicationInfo = (ApplicationInfo) event.hook();
-                if (me == null) {//race check
-                    initialiseInstanceDescriptor();
+                //Race condition double check
+                if (me == null) {
```

Comment:
https://en.wikipedia.org/wiki/Double-checked_locking#Usage_in_Java
https://www.javaworld.com/article/2074979/double-checked-locking--clever--but-broken.html
Page 348 of the book I'm about to plonk on your desk :)

Reply:
Oh, why did I miss that article? :)

Likely the easiest solution then is to replace any access to `me` with synchronized getter that can also initialize the descriptor when null.

Reply:
Is something like
`private synchronized InstanceDescriptorImpl getMyInstanceDescriptor() {
        if(me == null) {
            initialiseInstanceDescriptor();
        }
        return me;
    }`
the kind of thing you mean?